video_xblock/static/js/toggle-button.js

```diff
@@ -29,23 +29,55 @@ domReady(function() {
     },
     onClick: function onClick(event) {
       var menuItem = this.$$('.vjs-menu-item', this.el_.parentNode);
+      var el = event.currentTarget;
       Array.from(menuItem).forEach(function(caption) {
         caption.classList.remove('vjs-selected');
       });
-      var el = event.currentTarget;
       el.classList.add('vjs-selected');
-      this.player_.caption_lang = this.el_.dataset.lang;
-      this.player_.trigger('changelanguagetranscripts');
-    },
+
+      var tracks = this.player_.textTracks();
+      for (var i = 0; i < tracks.length; i++) {
+        var track = tracks[i];
+        if (track.kind === 'captions') {
+          this.player_.captionsLanguage = el.dataset.lang;
+          track.mode = track.language === this.player_.captionsLanguage ? 'showing': 'disabled';
+        }
+      }
+      this.player_.trigger('captionstrackchange');
+      this.player_.trigger('subtitlestrackchange');
+      this.player_.trigger('currentlanguagechanged');
+    }
   });
 
   var MenuButton = videojs.getComponent('MenuButton');
+  var ClickableComponent = videojs.getComponent('ClickableComponent');
+
   var ToggleButton = videojs.extend(MenuButton, {
     // base class for create buttons for caption and transcripts
     constructor: function constructor(player, options) {
       this.kind_ = 'captions';
-      MenuButton.call(this, player, options);
+
+      ClickableComponent.call(this, player, options);
+
+      if (!this.player_.singleton_menu) {
```

Comment:
Yes, two menu buttons (CC & Transcripts) share same menu list for language selection. Hence singleton.